chat-server/app/workflow/graph/single_dataset_graph/node/sql_agent.py

```
@@ -0,0 +1,81 @@
+)
+async def sql_agent(state: State, config: RunnableConfig) -> dict:
+    user_query = state.get("user_query", "") or ""
+    dataset_info = state.get("dataset_info", None)
+    prev_sql_queries = state.get("prev_sql_queries", [])
+    validation_result = state.get("validation_result", None)
+    query_result = state.get("query_result")
+
+    single_dataset_result = query_result.single_dataset_query_result
+
+    if not query_result or not single_dataset_result:
+        raise ValueError("query_result is not properly initialized")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)